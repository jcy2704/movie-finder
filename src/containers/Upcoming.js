import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { UpcomingMovies } from '../helpers/api_methods/api';
import { loadUpcoming } from '../actions';
import Loading from '../components/Loading';
import '../styles/catalogue/Catalogue.css';
import MoviePoster from '../components/MoviePoster';

const PopularCatalogue = ({ movies, loader }) => {
  const [isLoading, setLoading] = useState(true);

  useEffect(() => {
    UpcomingMovies(loader, setLoading);
  }, [loader]);

  if (isLoading) {
    return <Loading nothing />;
  }

  return (
    <ul className="catalogue-cont d-flex justify-content-center flex-wrap">
      {movies.map(movie => (
        <MoviePoster key={movie.id} movie={movie} />
      ))}
    </ul>
  );
};

PopularCatalogue.propTypes = {
  movies: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  loader: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({ movies: state.movies });

const mapDispatchToProps = dispatch => ({
  loader: movies => dispatch(loadUpcoming(movies)),
});

export default connect(mapStateToProps, mapDispatchToProps)(PopularCatalogue);
