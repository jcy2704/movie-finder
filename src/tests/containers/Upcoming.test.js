import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Upcoming from '../../containers/Upcoming';

const configureMockStore = configureStore();

const initialState = {
  movies: [
    {
      backdrop_path: '/inJjDhCjfhh3RtrJWBmmDqeuSYC.jpg',
      genre_ids: [
        28,
        878,
      ],
      id: 399566,
      overview: 'In a time when monsters walk the Earth, humanity’s fight for its future sets Godzilla and Kong on a collision course that will see the two most powerful forces of nature on the planet collide in a spectacular battle for the ages.',
      poster_path: '/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg',
      title: 'Godzilla vs. Kong',
      vote_average: 8.4,
    },
    {
      backdrop_path: '/inJjDhCjfhh3RtrJWBmmDqeuSYC.jpg',
      genre_ids: [
        28,
        878,
      ],
      id: 399566,
      overview: 'In a time when monsters walk the Earth, humanity’s fight for its future sets Godzilla and Kong on a collision course that will see the two most powerful forces of nature on the planet collide in a spectacular battle for the ages.',
      poster_path: '/pgqgaUx1cJb5oZQQ5v0tNARCeBp.jpg',
      title: 'Godzilla vs. Kong',
      vote_average: 8.4,
    },
  ],
  filter: [28],
};

const store = configureMockStore(initialState);

it('Renders Popular Movies', () => {
  const popular = renderer
    .create(
      <Provider store={store}>
        <Upcoming />
      </Provider>,
    )
    .toJSON();

  expect(popular).toMatchSnapshot();
});

describe('Popular movies details', () => {
  beforeEach(() => {
    const history = createMemoryHistory();

    render(
      <Provider store={store}>
        <Router history={history}>
          <Upcoming />
        </Router>
      </Provider>,
    );
  });

  it('loading', () => {
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });
});
