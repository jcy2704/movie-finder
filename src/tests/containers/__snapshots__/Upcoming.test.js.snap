// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Popular Movies 1`] = `
<div
  className="nothing"
>
  <div
    className="loading d-flex justify-content-center align-items-center"
  >
    <div
      className="spinner-grow text-warning"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
